src/integration/tests/firstTest.test.tsx

```diff
@@ -76,123 +150,498 @@ describe("First test", () => {
+
+    const currentDateInput = `${year}-${month}-${day}`;
+    const typeCateg = "Salário";
+    const msgTitle = "Encora";
+    const valorInput = "5000";
+
+    userEvent.clear(dataInputArea);
+
+    userEvent.type(dataInputArea, currentDateInput);
+    userEvent.selectOptions(categInput, typeCateg);
+    userEvent.type(titleInput, msgTitle);
+    userEvent.type(valueInput, valorInput);
+    userEvent.click(buttonAddItem);
 
-  it("should be possible to insert a revenue item with highlighted category and value in green.", () => {});
+    const tableArea = within(screen.getByTestId(TABLE_AREA_ID));
 
-  it("should be possible to insert an expense item with highlighted category and value in red.", () => {});
+    expect(tableArea.getByText(`R$ ${valorInput}`)).toHaveStyle("color: green");
+  });
 
-  it("sshould be able to calculate the balance by subtracting expenses from revenues.", () => {});
+  it("should be able to calculate the balance by subtracting expenses from revenues.", () => {
+    const now = new Date(Date.now());
+    const day = now.getDate().toString().padStart(2, "0");
+    const month = (now.getMonth() + 1).toString().padStart(2, "0");
+    const year = now.getFullYear();
 
-  it("should be able to update revenue and calculate the balance correctly.", () => {});
+    const inputArea = within(screen.getByTestId(INPUT_AREA_ID));
+    const dataInputArea = inputArea.getByTestId(INPUT_DATA_ITEM_ID);
+    const categInput = inputArea.getByTestId(INPUT_CATEG_ITEM_ID);
+    const titleInput = inputArea.getByTestId(INPUT_TITLE_ITEM_ID);
+    const valueInput = inputArea.getByTestId(INPUT_VALUE_ITEM_ID);
+    const buttonAddItem = inputArea.getByTestId(BUTTON_ADD_ITEM_ID);
+
+    const currentDateInput = `${year}-${month}-${day}`;
+
+    const typeCategRevenue = "Salário";
+    const msgTitleRevenue = "Encora";
+    const valorInputRevenue = "5000";
 
-  it("should be able to clear the data, categoria, titulo, and valor fields after adding an item.", () => {});
+    userEvent.clear(dataInputArea);
+
+    userEvent.type(dataInputArea, currentDateInput);
+    userEvent.selectOptions(categInput, typeCategRevenue);
+    userEvent.type(titleInput, msgTitleRevenue);
+    userEvent.type(valueInput, valorInputRevenue);
+    userEvent.click(buttonAddItem);
+
+    const typeCategExpense = "Aluguel";
+    const msgTitleExpense = "Apartamento";
+    const valorInputExpense = "800";
+
+    userEvent.type(dataInputArea, currentDateInput);
+    userEvent.selectOptions(categInput, typeCategExpense);
+    userEvent.type(titleInput, msgTitleExpense);
+    userEvent.type(valueInput, valorInputExpense);
+    userEvent.click(buttonAddItem);
+
+    const infoArea = within(screen.getByTestId(INFO_AREA_ID));
+    const balanceValue = infoArea.getByTestId(BALANCE_VALUE_ITEM_ID);
+
+    expect(balanceValue.textContent).toEqual(
+      `R$ ${parseFloat(valorInputRevenue) - parseFloat(valorInputExpense)}`
+    );
```

Comment:
Interesting! This is also correct, I'll just add here another way that you could do it:
(I didn't test by running, so it can have some issues rsrs)
```suggestion
    expect(infoArea.getByTestId(BALANCE_VALUE_ITEM_ID)).toHaveTextContent(
      `R$ ${parseFloat(valorInputRevenue) - parseFloat(valorInputExpense)}`
    );
```

Reply:
It worked, thanks! =)